### PR TITLE
Sys 1052/create projectitem model

### PR DESCRIPTION
**Pull request by kjallen**
Initial PR for ProjectItem essential fields. Please review the arguments on User related foreign keys, namely

```
models.ForeignKey(
        settings.AUTH_USER_MODEL,
        on_delete=models.SET_NULL,
        blank=True,
        null=True,
        related_name="+",
    )
```
The logical behavior should be as follows:
- Blank fields are allowed
- Null values are allowed
- If the user pointed to by this field is deleted, the result should be updated to null
- related_name was added to prevent reverse relationship from User table being created, not using this caused a conflict due to 2 User references in this table (created_by, late_modified by). Details at https://docs.djangoproject.com/en/4.1/ref/models/fields/#django.db.models.ForeignKey.related_name

Testing is similar to previous models:
- Run migrations, or restart containers
- Run `$ docker-compose exec django python manage.py loaddata --app oh_staff_ui seed-data` to import data
- Confirm content via admin ui

### CODE REVIEW

File: oh_staff_ui/models.py [hunk 1]
```diff
@@ -17,3 +18,34 @@ class ItemType(models.Model):
...
+    ark = models.CharField(max_length=40, blank=False, null=False)
+    create_date = models.DateField()
+    created_by = models.ForeignKey(
+        settings.AUTH_USER_MODEL,
+        on_delete=models.SET_NULL,
+        blank=True,
+        null=True,
+        related_name="+",
+    )
+    last_edit_date = models.DateField()
```
Comment (akohler) [inline oh_staff_ui/models.py:34]: @kjallen For consistency, please rename `last_edit_date` to `last_modified_date`.  This change will need to be made in `admin.py` and in the sample data as well.

File: oh_staff_ui/models.py [hunk 2]
```diff
@@ -17,3 +18,34 @@ class ItemType(models.Model):
...
+    last_modified_by = models.ForeignKey(
+        settings.AUTH_USER_MODEL,
+        on_delete=models.SET_NULL,
+        blank=True,
+        null=True,
+        related_name="+",
+    )
+    parent = models.ForeignKey("self", on_delete=models.SET_NULL, blank=True, null=True)
+    sequence = models.IntegerField(blank=True, null=True)
```
Comment (akohler) [inline oh_staff_ui/models.py:43]: Is it logical to have `sequence` be nullable?  If one item exists, it seems like it should have a `sequence` of `1`.  That also avoids possible sorting surprises: queries which order by sequence don't have to worry about NULL-first vs NULL-last behavior.

**Review by akohler — CHANGES_REQUESTED** [review]
@kjallen Most of my comments are questions: would it be better to...?  Otherwise, a few minor changes, mostly for consistency.  Thanks --Andy

File: oh_staff_ui/models.py (hunk 2 above)
Comment (kjallen) [reply oh_staff_ui/models.py:43]: Agreed, set to non-null and default of 1 if missing

File: oh_staff_ui/models.py [hunk 3]
```diff
@@ -17,3 +18,34 @@ class ItemType(models.Model):
...
+    )
+    last_edit_date = models.DateField()
+    last_modified_by = models.ForeignKey(
+        settings.AUTH_USER_MODEL,
+        on_delete=models.SET_NULL,
+        blank=True,
+        null=True,
+        related_name="+",
+    )
+    parent = models.ForeignKey("self", on_delete=models.SET_NULL, blank=True, null=True)
```
[the remark this answers is not in the document]
Comment (kjallen) [reply oh_staff_ui/models.py:42]: Yes, thank you for pointing out this use case -- it is the most common and one I was thinking of leaving this to null. It is perfectly acceptable to have workflow to re-assign children first, before deleting parent. Updated to `CASCADE`

File: oh_staff_ui/models.py [hunk 4]
```diff
@@ -17,3 +18,34 @@ class ItemType(models.Model):
 
     def __str__(self):
         return self.type
+
+
+class ProjectItem(models.Model):
+
+    ark = models.CharField(max_length=40, blank=False, null=False)
+    create_date = models.DateField()
+    created_by = models.ForeignKey(
```
[the remark this answers is not in the document]
Comment (kjallen) [reply oh_staff_ui/models.py:27]: Yes, and probably preferable given the traffic and user turnover (practically none) on this application. Updated to protect in `created_by` and `last_modified_by`

File: oh_staff_ui/models.py [hunk 5]
```diff
@@ -17,3 +18,34 @@ class ItemType(models.Model):
 
     def __str__(self):
         return self.type
+
+
+class ProjectItem(models.Model):
+
```
[the remark this answers is not in the document]
Comment (kjallen) [reply oh_staff_ui/models.py:24]: Unfortunately black is not smart enough to catch my inconsistencies :-)

**Review by akohler — APPROVED** [review]

**Pull request merged**